### ISSUE TITLE
PE tracking with cytoscapeviz.pl

### ISSUE THREAD

**Issue by ewilbanks** [MadsAlbertsen/multi-metagenome#1]
Hi there,

I've been working through your guide with my dataset and its been really useful - thanks for making such a great resource available!

I've run into a snag with the PE tracking step as my Illumina reads are in a different format and don't have a _1 or _2 underscore in the read header.  Looking at the entries in my sam file, it seems like read pairs have identical read headers, with all the pairing information stored in the SAM flags.

Do you by chance have another version of the script that could handle data like that, or a hankering to modify the script to choose pairs using SAM flags instead?  

Thanks!
Lizzy

**Comment by MadsAlbertsen** [comment]
Currently I do not have another version that works like that. But it is a very good suggestion that I will include in the next version of the script. 

Until then (hopefully soon) an easy fix is to rename the reads prior to mapping (or alternative rename the reads in the sam file).

However, I do remember that ctSkennerton made a similar script which I think uses the paired flag instead as you suggests. But I havn't tried it. You can find the script [here](https://github.com/ctSkennerton/scriptShed/blob/master/searchMappedPairs.py).

rgds
Mads

**Comment by MadsAlbertsen** [comment]
I just checked the code and cytoscapeviz.pl should in theory work as long as the readname is identical for your paired reads.

If you can send me an example of the readnames I can make a quick fix.

rgds
Mads

**Comment by ewilbanks** [comment]
Thanks!

Here's an example of two different sequencing runs that I'm using, w/ entries for a set of reads from both the fasta file and the sam file.  

One of my runs- 
FASTA:

> HWI-ST611:189:2:1101:1572:2248:0/1
> GGAGAAAGCGCGCTGGATCGAAGCGCGCAAAGACGAACTGCTGCCGGTGCCCTATTTTCATACCGTCTTCACGCTGCCGCATGCGATCAACGACTGGGTG
> HWI-ST611:189:2:1101:1572:2248:0/2
> GCCCCACCCAGTCGTTGATCGCATGCGGCAGCGTGAAGACGGTATGAAAATAGGGCACCGGCAGCAGTTCGTCTTTGCGCGCTTCGATCCAGCGCGCTTT

SAM:
HWI-ST611:189:2:1101:1572:2248:0        99      ctg7180000002145        81063   0       100M    =       81067   104     GGAGAAAGCGCGCTGGATCGAAGCGCGCAAAGACGAACTGCTGCCGGTGCCCTATTTTCATACCGTCTTCACGCTGCCGCATGCGATCAACGACTGGGTG    \*       NM:i:0  AS:i:100        XS:i:100
HWI-ST611:189:2:1101:1572:2248:0        147     ctg7180000002145        81067   0       100M    =       81063   -104    AAAGCGCGCTGGATCGAAGCGCGCAAAGACGAACTGCTGCCGGTGCCCTATTTTCATACCGTCTTCACGCTGCCGCATGCGATCAACGACTGGGTGGGGC    \*       NM:i:1  AS:i:99 XS:i:99

Another one of my runs-
FASTA:

> DJB775P1:392:D1R59ACXX:2:2107:15651:29274/1
> GCATCTTGCAGAACCGATTCAACCTTGTCGACAAGATATTGGAGGTCGATGGGTTTTTTTACAAAATCGATGGCTCCCATGTTCATAATTTTCTCAATGT
> DJB775P1:392:D1R59ACXX:2:2107:15651:29274/2
> CAATATAATTGACAAGCAAACTCCTGATCTTATTCTATTAGATCTTTTAATGCCCAAAATTAGCGGTTTCGATTTTCTCGAAGAAATACGTAAAAATGAA

SAM:
DJB775P1:392:D1R59ACXX:2:2211:2290:37940        99      100000045149    130     60      100M    =       474     430     GACCAAATATAGATTGAGATCAAGGTGATAAAATGGATCAGTCGTCTTAGTTGATACCGTCTTGCTGATCTTGGATTTGGCTTTTTTGATAACGGATTTG \*       NM:i:9  AS:i:55 XS:i:0
DJB775P1:392:D1R59ACXX:2:2211:2290:37940        147     100000045149    474     60      86M     =       130     -430    AGTATGTGATGTCAGAGACGCCCTCTGTCACGATCAGCGGCGCGGTAGGCGCGGCGAAATACTTGTAGAAGAGGAACTTTCGGTAC       \*       NM:i:9  AS:i:45 XS:i:0

**Comment by MadsAlbertsen** [comment]
From my small test with your SAM files it should work. What happens when you run the script?

**Comment by ewilbanks** [comment]
I actually just reparsed the read names to make them match your format which was pretty quick.  I did have to modify the script to handle the sam file output from bwa mem in that it had a \* in a field for unmapped reads which broke the script.  I just added an new statement at line 133 to force it to skip unmapped reads (see below).  After that it worked fine!
            if($splitline[2] !~ m/*/){     

Thanks!
